samples/test/downscoping-with-cab.test.js

```diff
@@ -0,0 +1,138 @@
+
+const {
+  GoogleAuth,
+  DownscopedClient,
+  OAuth2Client,
+} = require('google-auth-library');
+const {Storage} = require('@google-cloud/storage');
+const {assert} = require('chai');
+const {describe, it} = require('mocha');
+
+const bucketName = 'cab-int-bucket-brd3qlsuok';
+const objectName1 = 'cab-first-"brd3qlsuok.txt';
+const objectName2 = 'cab-second-"brd3qlsuok.txt';
+const CONTENT1 = 'first';
+
+describe('samples for downscoping with cab', () => {
+  const googleAuth = new GoogleAuth({
+    scopes: 'https://www.googleapis.com/auth/cloud-platform',
+  });
+  // Define the Credential Access Boundary object.
+  const cab = {
+    // Define the access boundary.
+    accessBoundary: {
+      // Define the single access boundary rule.
+      accessBoundaryRules: [
+        {
+          availableResource: `//storage.googleapis.com/projects/_/buckets/${bucketName}`,
+          // Downscoped credentials will have readonly access to the resource.
+          availablePermissions: ['inRole:roles/storage.objectViewer'],
+          // Only objects starting with the specified prefix string in the object name
+          // will be allowed read access.
+          availabilityCondition: {
+            expression:
+              "resource.name.startsWith('projects/_/buckets/" +
+              `${bucketName}/objects/${objectName1}')`,
+          },
+        },
+      ],
+    },
+  };
+
+  it('should only download file for object with cab permission', async () => {
+    const projectId = await googleAuth.getProjectId();
+    const errorMessage =
+      'does not have storage.objects.get access to the Google Cloud Storage object.';
+
+    // Obtain an authenticated client via ADC.
+    const client = await googleAuth.getClient();
+
+    // Use the client to create a DownscopedClient.
+    const cabClient = new DownscopedClient(client, cab);
+
+    // Create the OAuth credentials (the consumer).
+    const oauth2Client = new OAuth2Client();
+    // We are defining a refresh handler instead of a one-time access
+    // token/expiry pair.
+    // This will allow the consumer to obtain new downscoped tokens on
+    // demand every time a token is expired, without any additional code
+    // changes.
+    oauth2Client.refreshHandler = async () => {
+      // The common pattern of usage is to have a token broker pass the
+      // downscoped short-lived access tokens to a token consumer via some
+      // secure authenticated channel. For illustration purposes, we are
+      // generating the downscoped token locally.
+      const refreshedAccessToken = await cabClient.getAccessToken();
+      return {
+        access_token: refreshedAccessToken.token,
+        expiry_date: refreshedAccessToken.expirationTime,
+      };
+    };
+
+    const storageOptions = {
+      projectId,
+      authClient: {
+        sign: () => {
+          Promise.reject('unsupported');
+        },
+        getCredentials: async () => {
+          Promise.reject();
```

Comment:
Promise.reject('unsupported')?

Reply:
This one is following our agreement on designs [doc](https://docs.google.com/document/d/1CyFdTAxe6EIcbReSVbsqS8IKYpzeweuA5OAtCAsF5fg/edit)